helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+        String formatted = format(name);
+        return Arrays.stream(LEGACY_VALUES).anyMatch(mat -> parseLegacyVersionMaterialName(mat).equals(formatted));
+    }
+
+    /**
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    public static XMaterial matchXMaterial(@Nonnull Material material) {
+        return matchXMaterial(material.name());
+    }
+
+    /**
+     * Can also be used like: <b>MATERIAL:DATA</b>
+     * E.g. <b>RED_WOOL:14</b> or <b>WOOL:14</b>
+     *
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    public static XMaterial matchXMaterial(@Nonnull String name) {
+        // -1 Determines whether the item's data is unknown and only the name is given.
+        // Checking if the item is damageable won't do anything as the data is not going to be checked in requestOldMaterial anyway.
+        return name.contains(":") ? matchXMaterial(name.substring(0, name.indexOf(":")),
+                Byte.parseByte(StringUtils.replace(name.substring(name.indexOf(":") + 1), " ", "")))
+                : matchXMaterial(name, (byte) -1);
+    }
+
+    /**
+     * Parses the material name and data argument as a {@link Material}.
+     *
+     * @param name name of the material
+     * @param data data of the material
+     */
+    @Nullable
+    public static Material parseMaterial(@Nonnull String name, byte data) {
+        return matchXMaterial(name, data).parseMaterial();
+    }
+
+    /**
+     * @param item the ItemStack to match its material and data.
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    @SuppressWarnings("deprecation")
+    public static XMaterial matchXMaterial(@Nonnull ItemStack item) {
+        Objects.requireNonNull(item, "ItemStack cannot be null");
+        return isDamageable(item.getType().name()) ? matchXMaterial(item.getType().name(), (byte) 0) :
+                matchXMaterial(item.getType().name(), (byte) item.getDurability());
+    }
+
+    /**
+     * Matches the argument string and its data with a XMaterial.<br>
+     *
+     * @param name the name of the material
+     * @param data the data byte of the material
+     * @return a XMaterial from the enum (with the same legacy name and data if in older versions.)
+     */
+    @Nonnull
+    public static XMaterial matchXMaterial(@Nonnull String name, byte data) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        name = format(name);
+
+        if ((contains(name) && data <= 0) && (isNewVersion() || !isDuplicated(name)))
+            return valueOf(name);
+
+        // TODO Temporary but works - Please find a more reasonable fix for duplicated materials.
+        if (isDuplicated(name) && !isNewVersion())
+            return requestDuplicatedXMaterial(name, data);
+
+        return requestOldXMaterial(name, data);
+    }
+
+    /**
+     * Gets the XMaterial based on the Material's ID and data.<br>
+     * You should avoid using this for performance reasons.
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.